File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/adapter/AdapterPattern.java
package jv.com.tencent.cain.design_pattern.adapter;

/**
 * @author cainjiang
 * @date 2020/4/15
 */
public class AdapterPattern {

    public static void main(String[] args) {
        Player forwardPlayer = new ForwardPlayer("Battier");
        Player guardPlayer = new GuardPlayer("McGrady");
        TranslateAdapter centerPlayer = new TranslateAdapter("Translator");
        centerPlayer.setCenterPlayer(new ForeignCenterPlayer("Yao Ming"));
        System.out.println("---start play---");
        forwardPlayer.attack();
        guardPlayer.defense();
        centerPlayer.attack();
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/adapter/AdapterPattern.kt
package kt.com.tencent.cain.desigin_pattern.adapter

/**
 * @author cainjiang
 * @date 2020/4/15
 */
fun main() {
    val forwardPlayer = ForwardPlayer("Battier")
    val guardPlayer = GuardPlayer("McGrady")
    val centerPlayer = TranslateAdapter("Translator", ForeignCenterPlayer("Yao Ming"))
    println("--- start play ---")
    forwardPlayer.attack()
    guardPlayer.defense()
    centerPlayer.defense()
}